src/core/search/sort_indices.cc

```diff
@@ -66,18 +71,23 @@ template <typename T> PMR_NS::memory_resource* SimpleValueSortIndex<T>::GetMemRe
 template struct SimpleValueSortIndex<double>;
 template struct SimpleValueSortIndex<PMR_NS::string>;
 
-double NumericSortIndex::Get(DocId id, DocumentAccessor* doc, std::string_view field) {
+std::optional<double> NumericSortIndex::Get(DocId id, DocumentAccessor* doc,
+                                            std::string_view field) {
   auto str = doc->GetStrings(field);
```

Comment:
`str` is ambigous clear. it is a vector? either name it `str_vec` or provide an explicit type